Protocol/Tests/Protocol/Params/Param/Interprete/Exceptions/Exception/Display/CheckStateAttribute.cs

```diff
@@ -0,0 +1,155 @@
+
+                    if (stateValue == EnumDisplayState.Enabled)
+                    {
+                        results.Add(Error.UnrecommendedEnabledValue(this, param, state, param.Id?.RawValue).WithExtraData(ExtraData.ExceptionToFix, exception));
+                        continue;
+                    }
+
+                    // Untrimmed
+                    if (status.HasFlag(GenericStatus.Untrimmed))
+                    {
+                        results.Add(Error.UntrimmedAttributeValue(this, param, state, param.Id?.RawValue, rawDisplayValue).WithExtraData(ExtraData.ExceptionToFix, exception));
+                        continue;
+                    }
+                }
+            }
+
+            return results;
+        }
+
+        public ICodeFixResult Fix(CodeFixContext context)
+        {
+            CodeFixResult result = new CodeFixResult();
+            if (!(context.Result.ReferenceNode is IParamsParam readParam))
+            {
+                result.Message = "context.Result.Node is not of type IParamsParam";
+                return result;
+            }
+
+            var editParam = context.Protocol?.Params?.Get(readParam);
+            if (editParam == null)
+            {
+                result.Message = "editParam is null.";
+                return result;
+            }
+
+            if (!context.Result.ExtraData.TryGetValue(ExtraData.ExceptionToFix, out object o)
+                || !(o is IParamsParamInterpreteExceptionsException readException))
+            {
+                result.Message = "ExtraData[exceptionToFix] is not of type IParamsParamInterpreteExceptionsException";
+                return result;
+            }
+
+            var editException = editParam.Interprete?.Exceptions?.Get(readException);
+            if (editException == null)
+            {
+                result.Message = "editException is null.";
+                return result;
+            }
+
+            if (editException.Display == null)
+            {
+                result.Message = "editException.Display is null.";
+                return result;
+            }
+
+            switch (context.Result.ErrorId)
+            {
+                case ErrorIds.EmptyAttribute:
+                case ErrorIds.UnrecommendedEnabledValue:
+                case ErrorIds.InvalidAttributeValue:
+                    editException.Display.State.Value = EnumDisplayState.Disabled;
+                    result.Success = true;
+                    break;
+
+                case ErrorIds.MissingAttribute:
+                    editException.Display.State = new ParamsParamInterpreteExceptionsExceptionDisplayState(EnumDisplayState.Disabled);
+                    result.Success = true;
+                    break;
+
+                case ErrorIds.UntrimmedAttributeValue:
+                    Enum.TryParse<EnumDisplayState>(editException.Display.State.RawValue.Trim() == "disabled" ? "Disabled" : "Enabled", out EnumDisplayState parsedValue);
```

Comment:
You can use the EnumDisplayStateConverter to convert to and from the enums in a protocol.

Reply:
Fixed